SYCL/ESIMD/api/functional/ctors/ctor_fill.hpp

```diff
@@ -0,0 +1,226 @@
+          [=]() SYCL_ESIMD_KERNEL {
+            TestCaseT::template call_simd_ctor<DataT, NumElems>(
+                base_value, step_value, out);
+          });
+    });
+
+    return result;
+  }
+};
+
+// Constructing verbose failure message.
+template <int NumElems>
+void on_failure(const std::string &step_value, const std::string &base_value,
+                const std::string &elem_from_ref_data,
+                const std::string &elem_from_result_data,
+                const std::string &data_type,
+                const std::string &test_case_description) {
+  log::fail<NumElems>("Simd by " + test_case_description +
+                          " failed, retrieved: " + elem_from_result_data +
+                          ", expected: " + elem_from_ref_data +
+                          ", with step value: " + step_value +
+                          ", with base value: " + base_value,
+                      data_type);
+}
+
+// Struct for verifying test result, failed test if a retrived value is not
+// equal to an expected value.
+template <init_val BaseVal, init_val Step> struct verify_obtained_result {
+  template <typename DataT, int NumElems>
+  bool verify_test_result(const std::vector<DataT> &ref_data,
+                          const esimd_tests::shared_vector<DataT> &result_data,
+                          const std::string &data_type,
+                          const std::string &test_case_description) {
+    bool pass{true};
+    DataT step_value{get_base_value<DataT, Step>()};
+    DataT base_value{get_base_value<DataT, BaseVal>()};
+    size_t zero_elem{0};
+
+    if (!are_bitwise_equal(ref_data[zero_elem], base_value)) {
+      pass = false;
+      on_failure<NumElems>(std::to_string(step_value),
+                           std::to_string(base_value),
+                           std::to_string(ref_data[zero_elem]),
+                           std::to_string(result_data[zero_elem]), data_type,
+                           test_case_description);
+    }
+    for (size_t i = 1; i < ref_data.size(); ++i) {
+      if (!are_bitwise_equal(
+              result_data[i],
+              static_cast<DataT>(result_data[i - 1] + step_value))) {
+        if constexpr (type_traits::is_sycl_floating_point_v<DataT>) {
+          if (std::isnan(result_data[i]) &&
+              (std::isnan(step_value) || std::isnan(base_value))) {
+            continue;
+          }
+        }
+        pass = false;
+        on_failure<NumElems>(
+            std::to_string(step_value), std::to_string(base_value),
+            std::to_string(result_data[i - 1] + step_value),
+            std::to_string(result_data[i]), data_type, test_case_description);
+      }
+    }
+
+    return pass;
+  }
+};
+
+// Iterating over provided types and dimensions, running test for each of them.
+template <typename TestT, init_val BaseVal, init_val Step, typename... Types,
+          int... Ints>
+bool run_verifying(sycl::queue &queue,
```

Comment:
```suggestion
bool run_verification(sycl::queue &queue,
```
Aren't you using the `run_verification_with_chosen_test_type` name for the similar purpose in `ctor_default.cpp`?

Reply:
this comment was applied [here](https://github.com/intel/llvm-test-suite/pull/598/commits/652dcd3b7c72dceecf81bb7bd486398d8664c24d#diff-7fc90b589428e6e89076f07f5ff280a217b7f5c8e836bfe6c742734825074ef4R203)